gensim/similarity_learning/drmm_tks.py

```diff
@@ -0,0 +1,111 @@
+import keras.backend as K
+from keras.models import Model
+from keras.layers import Input, Embedding, Dot, Dense, Lambda, Reshape, Dropout
+from keras.activations import softmax
+
+
+class DRMM_TKS:
+    """This is a variant version of DRMM, which applied topk pooling in the matching matrix.
+    It has the following steps:
+    1. embed queries into embedding vector named 'q_embed' and 'd_embed' respectively
+    2. computing 'q_embed' and 'd_embed' with element-wise multiplication
+    3. computing output of upper layer with dense layer operation
+    4. take softmax operation on the output of this layer named 'g' and find the k largest entries named 'mm_k'.
+    5. input 'mm_k' into hidden layers, with specified length of layers and activation function
+    6. compute 'g' and 'mm_k' with element-wise multiplication.
+
+    # Returns
+        Score list between queries and documents.
+    """
+
+    def __init__(self, embedding, embed_dim, vocab_size, embed_trainable=False, target_mode='ranking',
+                 topk=20, dropout_rate=0., text_maxlen=40, hidden_sizes=[5, 1]):
```

Comment:
`text_maxlen` default is too naive (why so small?)

Reply:
Seeing the general queries are around 10 to 15 words, I felt this would do sufficient.
Setting it to 100 for now.

Reply:
in general case, query and answer can be 2 large document (not classic short-query)